site/app/libraries/database/PostgresqlDatabaseQueries.php

```diff
@@ -1016,16 +1033,42 @@ public function getGradeableDataAll(\app\models\gradeable\Gradeable $gradeable,
         if(count($users) > 0) {
             $user_placeholders = implode(',', array_fill(0, count($users), '?'));
             $selector_union_list[] = "(gd.gd_user_id IN ($user_placeholders))";
+            $selector_union_list[] = "(egd.user_id IN ($user_placeholders))";
+
+            // Get all the teams each user is on
+            $subquery = "SELECT team_id FROM teams WHERE user_id IN ($user_placeholders)";
+            $this->course_db->query($subquery, $users);
+            $teams = array_merge($teams, array_map(function ($row) {
+                return $row['team_id'];
+            }, $this->course_db->rows()));
         }
         // Teams were provided, so check that list
         if(count($teams) > 0) {
             $team_placeholders = implode(',', array_fill(0, count($teams), '?'));
             $selector_union_list[] = "(gd.gd_team_id IN ($team_placeholders))";
+            $selector_union_list[] = "(egd.team_id IN ($team_placeholders))";
+        }
+
+        $selector_intersection_list = [];
+        //
+        // Generate selector for the gradeables the user wants
+        //
+        $gradeable_placeholders = implode(',', array_fill(0, count($gradeables_by_id), '?'));
+        $selector_intersection_list[] = "(gd.g_id IN ($gradeable_placeholders) OR egd.g_id IN ($gradeable_placeholders))";
+
+        // Add the user selector later so the gradeable selector can be first
+        if(count($selector_union_list) > 0) {
+            $selector_intersection_list[] = '(' . implode(' OR ', $selector_union_list) . ')';
         }
-        $submitter_selector = implode(' OR ', $selector_union_list);
+
+        // Create the complete selector
+        $selector = implode(' AND ', $selector_intersection_list);
 
         $query = "
             SELECT /* Select everything we retrieved */
+            
+              (CASE WHEN gd.g_id IS NULL THEN egd.g_id ELSE gd.g_id END) as g_id,
+            
               /* Gradeable Data */
               gd.gd_id AS id,
               gd.gd_user_id as user_id,
```

Comment:
If the `gradeable_data` is null then user_id will be null yet you still have user data from the `users` table `u` below. This means you'll get a user with a name and info but no id.